### PR TITLE
CORE-8638 Move getCpiMetadata, getCpkMetadata, getNumberOfComponentGroups, getSchemaVersion from TransactionMetadata to impl

### PR DESCRIPTION
**Pull request by vlajos**
CORE-8638 Move getCpiMetadata, getCpkMetadata, getNumberOfComponentGroups, getSchemaVersion from TransactionMetadata to impl

API: https://github.com/corda/corda-api/pull/915

### CODE REVIEW

**Review by MrMatthewLayton — DISMISSED** [review]
LGTM (although, I have no context as to why this info is now being internalised)

**Review by mnesbit — DISMISSED** [review]
OK

File: libs/ledger/ledger-common-data/src/main/kotlin/net/corda/ledger/common/data/transaction/TransactionMetadataImpl.kt [hunk 1]
```diff
@@ -6,7 +6,8 @@ import net.corda.v5.ledger.common.transaction.CordaPackageSummary
 import net.corda.v5.ledger.common.transaction.TransactionMetadata
 
 @CordaSerializable
-class TransactionMetadataImpl (private val properties: Map<String, Any>) : TransactionMetadata {
+class TransactionMetadataImpl(private val properties: Map<String, Any>) : TransactionMetadata,
+    TransactionMetadataInternal {
```
Comment (blsemo) [inline libs/ledger/ledger-common-data/src/main/kotlin/net/corda/ledger/common/data/transaction/TransactionMetadataImpl.kt]: Should this inherit both from `TransactionMetadata` and `TransactionMetadataInternal`?

**Review by blsemo — COMMENTED** [review]
Looks fine to me, apart from the inheritance question.

File: libs/ledger/ledger-common-data/src/main/kotlin/net/corda/ledger/common/data/transaction/TransactionMetadataImpl.kt (hunk 1 above)
Comment (vlajos) [reply libs/ledger/ledger-common-data/src/main/kotlin/net/corda/ledger/common/data/transaction/TransactionMetadataImpl.kt]: `TransactionMetadataInternal` should be enough. I have just updated it.

**Review by mnesbit — APPROVED** [review]
OK

**Pull request merged**